File: backend/app/schemas/admin_session.py
"""管理员会话查看 API 的 Pydantic 模型。参照 BACKEND_STRUCTURE §2.10。"""

from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.schemas.session import SessionMessageOut


class AdminSessionListItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    username: str = ""
    story_id: int
    story_title: str = ""
    story_version_id: int
    rag_config_id: int
    mode: str
    status: str
    turn_count: int
    opening_goal: str
    created_at: datetime | None = None
    updated_at: datetime | None = None


class TranscriptSessionMeta(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    story_id: int
    story_version_id: int
    rag_config_id: int
    mode: str
    status: str
    turn_count: int
    opening_goal: str


class TranscriptResponse(BaseModel):
    session: TranscriptSessionMeta
    messages: list[SessionMessageOut]


class AdminUserFeedbackOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    session_id: int
    message_id: int
    feedback_type: str
    content: str | None
    reviewed: bool
    created_at: datetime | None = None


class FeedbackListResponse(BaseModel):
    items: list[AdminUserFeedbackOut]


class AdminSessionsListResponse(BaseModel):
    items: list[AdminSessionListItem]
    total: int

File: frontend/src/types/adminSession.ts
/** 与后端 app/schemas/admin_session.py 对齐 */

import type { SessionMessage } from "./session";

export interface AdminSessionListItem {
  id: number;
  user_id: number;
  username: string;
  story_id: number;
  story_title: string;
  story_version_id: number;
  rag_config_id: number;
  mode: string;
  status: string;
  narrative_status: string;
  narrative_plan: Record<string, unknown>;
  turn_count: number;
  opening_goal: string;
  created_at?: string | null;
  updated_at?: string | null;
}

export interface AdminSessionsListResponse {
  items: AdminSessionListItem[];
  total: number;
}

export interface TranscriptSessionMeta {
  id: number;
  user_id: number;
  story_id: number;
  story_version_id: number;
  rag_config_id: number;
  mode: string;
  status: string;
  narrative_status: string;
  narrative_plan: Record<string, unknown>;
  turn_count: number;
  opening_goal: string;
}

export interface TranscriptResponse {
  session: TranscriptSessionMeta;
  messages: SessionMessage[];
}

export interface AdminUserFeedbackOut {
  id: number;
  session_id: number;
  message_id: number;
  feedback_type: string;
  content: string | null;
  reviewed: boolean;
  created_at?: string | null;
}

export interface FeedbackListResponse {
  items: AdminUserFeedbackOut[];
}
